crates/ty_python_semantic/src/place.rs

```diff
@@ -763,7 +763,7 @@ fn place_by_id<'db>(
                 .expr
                 .is_name_and(|name| matches!(name, "__slots__" | "TYPE_CHECKING"));
 
-            if scope.file(db).is_stub(db) {
+            if scope.file(db).is_stub(db) || scope.scope(db).visibility().is_private() {
```

Comment:
This is one of the key changes here. Can we please update the comment below to also explain why we do not union with Unknown when the scope visibility is private?